src/Infrastructure/Persistence/Doctrine/Repository/Regulation/RegulationOrderRecordRepository.php

```diff
@@ -191,16 +191,44 @@ public function convertToCifsPolylines(string $geometry): array
                     SELECT (components.dump).geom AS geom FROM (
                         SELECT ST_Dump(ST_LineMerge(:geom)) AS dump
                     ) AS components
+                ), linestring_buffered AS (
+                    SELECT (
+                        CASE
+                        -- Waze requires at least 2 points, and a linestring length of at least 35 meters.
+                        -- In case we have a POINT (for example noEntry on a single house number), we take
+                        -- a buffer of the road around this point.
+                        WHEN (SELECT COUNT(*) FROM ST_DumpPoints(linestring.geom)) = 1
+                            THEN ST_Intersection(
+                                ST_SetSRID(:road_geometry::geometry, 4326),
+                                ST_Transform(
+                                    ST_Buffer(
+                                        ST_Transform(
+                                            ST_SetSRID((SELECT geom FROM ST_DumpPoints(linestring.geom) LIMIT 1), 4326),
+                                            2154
+                                        ),
+                                        35
+                                    ),
+                                    4326
+                                )
```

Comment:
Si jamais il y a une façon moins verbeuse de prendre un buffer en mètres, je prends

(En effet le système de coordonnées 4326 est en degrés et pas en mètres)

Reply:
Est-ce qu'on ne ferait pas une conversion `mètres` vers `degrés °` en considérant la Terre comme étant une sphère parfaite de 6371 km de diamètre ?
En utilisant ce calculateur : https://lucidar.me/fr/online-unit-converter-length-to-angle/convert-degrees-to-meters/ j'obtiens `0.0003147625620715557°`

Reply:
Sinon : pour obtenir le premier point de la `Line` : au lieu de `SELECT geom FROM ST_DumpPoints(linestring.geom) LIMIT 1` on peut utiliser `ST_PointN` : `SELECT ST_PointN(linestring.geom, 1) AS geom` (attention, "1-based index", la numérotation commence à 1)

Reply:
J'ai écrit degrés mais en fait c'est plutôt "en degrés Mercator (4326)"

Or la projection du Mercator n'est pas un système de coordonnées sphériques uniforme, une même distance va correspondre à un nombre de degrés différent selon la latitude/longitude...

Reply:
Oui effectivement, j'ai testé dans QGIS avec l'extension WKT et ça ne se superpose pas parfaitement.

Reply:
Le ST_PointN() ne fonctionne pas car dans ce cas la géométrie est un POINT, or ST_PointN ne fonctionne que sur des lignes

J'ai gardé le LIMIT 1 pour l'instant...